modal/functions.py

```diff
@@ -355,6 +359,11 @@ async def pump_inputs():
             logger.debug(
                 f"Pushing {len(items)} inputs to server. Num queued inputs awaiting push is {input_queue.qsize()}."
             )
+
+            # Wait for inputs to be finished before continuing, as server will deny put inputs request
+            while len(pending_outputs) - len(completed_outputs) + len(items) > MAP_INVOCATION_CONCURRENCY:
+                await asyncio.sleep(0.5)
```

Comment:
This could potentially "lock up" the client without any feedback to the user in case nothing is consumed from the map in a long time (or at all). Perhaps add some logging at a certain time interval, saying something like "Too many queued inputs to the map - waiting for caller to consume outputs before adding additional inputs" to help the user understand what's happening?

Reply:
I think this could be better integrated into the progress bar potentially.

A notion of completed inputs, uploaded inputs and total inputs?